Source/Core/CesiumTerrainProvider.js

```diff
@@ -601,6 +603,18 @@ define([
         },
 
         /**
+         * Gets a promise that resolves to true when the provider is ready for use.
+         * @memberof CesiumTerrainProvider.prototype
+         * @type {Promise.<Boolean>}
+         * @readonly
+         */
+        readyPromise : {
+            get : function() {
+                return this._readyPromise;
```

Comment:
Since we are using a deferred, this should be `this._readyPromise.promise` see #3152 for why.